ansible_base/lib/abstract_models/common.py

```diff
@@ -72,22 +72,34 @@ def _attributable_user(self, warn_nonexistent_system_user):
 
     def save(self, *args, warn_nonexistent_system_user=True, **kwargs):
         update_fields = list(kwargs.get('update_fields', []))
-        user = self._attributable_user(warn_nonexistent_system_user)
 
         # Manually perform auto_now_add and auto_now logic.
         now = timezone.now()
-        if not self.pk and not self.created_on:
-            self.created_on = now
-            self.created_by = user
-            if 'created_on' not in update_fields:
-                update_fields.append('created_on')
-            if 'created_by' not in update_fields:
+
+        user = self._attributable_user(warn_nonexistent_system_user)
+        # Its possible for _attributable_user to return None in which case we can't save the created/modified by fields so warn
+        if not user:
+            logger.error("Unable to set create/modified by attributes on {self.__class__}/{self.pk} because user was not set. Save will proceed")
+        else:
+            if 'created_by' not in update_fields or not self.created_by:
+                self.created_by = user
                 update_fields.append('created_by')
+                if not self.pk:
+                    logger.warning(f"Existing model {self.__class__}/{self.pk} needed created_on updated!")
+
+            if 'modified_by' not in update_fields or not self.modified_by:
+                self.modified_by = user
+                update_fields.append('modified_by')
+
+        if 'created_on' not in update_fields or not self.created_on:
```

Comment:
should it be allowed to change "created_on"?

Reply:
I see an argument that some system-level logic may need to update `created_by` after the fact. Imagine it has to create an object before it knows the attributable user.

But that argument doesn't apply to `created_on`, because there's no valid reason you wouldn't be able to get a timestamp when creating the object.

Reply:
Yea, this should be updated now.